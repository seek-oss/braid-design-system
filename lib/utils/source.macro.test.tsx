import pluginTester from 'babel-plugin-tester';
import plugin from 'babel-plugin-macros';

pluginTester({
  plugin,
  pluginName: 'source.macro',
  snapshot: true,
  tests: {
    'smoke test': {
      babelOptions: { filename: __filename },
      code: `
      import source from './source.macro'

      const result = source({
        // comment
        hello: () => 'world'
      });
    `,
    },
  },
});
