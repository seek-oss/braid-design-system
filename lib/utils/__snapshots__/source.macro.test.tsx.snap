// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`source.macro smoke test: smoke test 1`] = `

import source from './source.macro'

const result = source({
  // comment
  hello: () => 'world'
});

      ↓ ↓ ↓ ↓ ↓ ↓

const result = {
  code: "{\\n  // comment\\n  hello: () => 'world'\\n}",
  value: {
    // comment
    hello: () => 'world',
  },
};


`;
